pkg/scheduler/objects/application.go

```diff
@@ -1640,9 +1645,32 @@ func (sa *Application) removeAllocationInternal(uuid string, releaseType si.Term
 		}
 	}
 	delete(sa.allocations, uuid)
+	if alloc.GetPriority() == sa.allocMinPriority {
+		sa.updateAllocationMinPriority(alloc.GetPriority())
+	}
 	return alloc
 }
 
+func (sa *Application) updateAllocationMinPriority(allocPriority int32) {
```

Comment:
This can just be a simple loop, tracking the lowest priority value we find. The extra minPriorityCount stuff doesn't save any time and adds complexity.